### PR TITLE
insert the information of the columns in which all the values are null to the meta table

### PR DESCRIPTION
**Pull request by aojwang**
JIRA: [MADLIB-635](http://jira.madlib.net/browse/MADLIB-635)
During encoding process, if all the values of a column is null, then we will remove the column from breakup table in 'ignore' mode. Therefore, besides insert the columns' information in the breakup table to the metatable, we need to insert those columns in which all the values are null to the meta table. This PR is used to track the issue.

When i fix this bug, I found that we didn't change the file Module.yml. As now we invoke some functions from array_ops, we need to depend on that module. As it's a minor problem, i combine to this PR. Thanks.

### CODE REVIEW

File: methods/cart/src/pg_gp/dt_preproc.sql_in [hunk 1]
```diff
@@ -1633,6 +1633,29 @@ m4_changequote(>>>`<<<, >>>'<<<)
...
+                 SELECT id, (ARRAY[''%''])[id] as column_name,
+                        ''f'' as column_type, ''t'', NULL, 0
+                 FROM (
+                     SELECT generate_series(1, %) id
+                     EXCEPT
+                     SELECT id FROM % WHERE column_type = ''f'' 
+                 ) t',
+                ARRAY[
+                    meta_tbl_name,
+                    array_to_string(attr_col_names, ''','''),
```
Comment (haradh1) [inline methods/cart/src/pg_gp/dt_preproc.sql_in]: Do we care empty  or NULL array?

Comment (cwelton) [inline methods/cart/src/pg_gp/dt_preproc.sql_in]: The `array_to_string(attr_col_names, ''',''')` probably doesn't work correctly with column names that require quote_ident(), however I strongly suspect that this is just one of many for the DT code, so I doubt that breaks anything new.

Comment (aojwang) [inline methods/cart/src/pg_gp/dt_preproc.sql_in]: It should not be null here. Null means no features. I add the check in function "MADLIB_SCHEMA.__check_training_table" before encoding.

We have 4 places need to use quote_ident and quote_literal. All of places were changed. I use the following table to do the test, it works well for training, classification, score and display. Please help double check the correctness of the changes. If I am wrong, please correct me. Thanks.

``` sql
testdb=# \d golf_2
           Table "public.golf_2"
   Column    |       Type       | Modifiers 
-------------+------------------+-----------
 id          | integer          | 
 out,look    | text             | 
 temperature | double precision | 
 humi'dity   | double precision | 
 win'dy      | text             | 
 class       | text             | 
Distributed by: (id)

```

File: src/config/Modules.yml [hunk 2]
```diff
@@ -15,6 +15,7 @@ modules:
     - name: data_profile
       depends: ['sketch']
     - name: cart
+      depends: ['array_ops']
```
Comment (haradh1) [inline src/config/Modules.yml]: Do we need this?

**Comment by cwelton** [comment]
Looks good.

File: src/config/Modules.yml (hunk 2 above)
Comment (aojwang) [inline src/config/Modules.yml]: Sorry. It's my mistake. I double check with Yi and the src code. We needn't this anymore. Done. Thanks.

**Pull request closed without merging**